domain/src/main/java/org/akvo/flow/domain/interactor/GetSavedDataPoints.java

```diff
@@ -86,23 +87,27 @@ protected GetSavedDataPoints(ThreadExecutor threadExecutor,
     }
 
     private Observable<List<DataPoint>> filterDataPoints(@NonNull List<DataPoint> dataPoints,
-            final String filter) {
+            @NonNull final String filter) {
         return Observable
                 .fromIterable(dataPoints)
                 .filter(new Predicate<DataPoint>() {
                     @Override
                     public boolean test(@NonNull DataPoint dataPoint) throws Exception {
                         String name = dataPoint.getName();
-                        if (name != null) {
-                            name = name.toLowerCase();
-                        }
                         String id = dataPoint.getId();
-                        return ((name != null && name
-                                .contains(filter.toLowerCase()))
-                                || (id != null && id.startsWith(filter)));
```

Comment:
Regarding the `id`, I understand that because you dropped this, the search for identifier will also include those where the identifier characters may occur in the middle of the identifier string?  Is this an intentional choice?

Reply:
yes, Jana asked for the id to be found even if the string searched is not at the beginning of the id but also in the middle (same behaviour as for name)